Add notes about release issues to Plugin Release Process (#487)
mattermost/mattermost-handbook: operations/research-and-development/engineering/plugin-release-process.md

  * Documentation through bundled releases aren’t viewable through the app and users will be looking at the latest master commit in the GitHub repo.
* The PR for the version bump (in the plugin repo) does not mean that is the last commit to get tagged. This step only bumps the version. The tagging step actually determines the commit that is tagged with the release tag.

### Create release issue

* Create an issue on the corresponding repository to track the whole release process.
  * **Title:** `Release vX.Y.Z`

>   * **Summary:** Add every outstand PR or issue as an item on as [taks list](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-task-lists).

Reviewer: ```suggestion
  * **Summary:** Add every outstanding PR or issue as an item on as [task list](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-task-lists).
```

>   * **Summary:** Add every outstanding PR or issue as an item on as [task list](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-task-lists).

Reviewer 2: ```suggestion
  * **Summary:** Add every outstanding PR or issue as a [task list item](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-task-lists).
```

> * Add that issue to the [Plugin Release board](https://github.com/orgs/mattermost/projects/2) in `Code Changes Needed`.

Reviewer 2: ```suggestion
* Add that issue to the [Plugin Releases board](https://github.com/orgs/mattermost/projects/2) in `Code Changes Needed`.
```
For consistency with the actual board name.
